### PR TITLE
Replace solve with solveset in sympy.ntheory

### PR DESCRIPTION
**Pull request by aktech**
TODO:
- [x] `continued_fraction.py`

Partial Fix for #8711

@hargup @flacjacket

### CODE REVIEW

File: sympy/ntheory/continued_fraction.py [hunk 1]
```diff
@@ -154,7 +155,7 @@ def untillist(cf):
 
     if period:
         y = Dummy('y')
-        solns = solve(continued_fraction_reduce(period + [y]) - y, y)
+        solns = list(solveset(continued_fraction_reduce(period + [y]) - y, y))
```
Comment (moorepants) [inline sympy/ntheory/continued_fraction.py]: Please don't remove the solve test.

Comment (aktech) [inline sympy/ntheory/continued_fraction.py]: This is not a test.

Comment (moorepants) [inline sympy/ntheory/continued_fraction.py]: Sorry, I was just looking through all of your changes on these PRs.

Comment (moorepants) [inline sympy/ntheory/continued_fraction.py]: I may have done the same thing on the other PRs.

Comment (aktech) [inline sympy/ntheory/continued_fraction.py]: I would revert all the Merged & Unmerged `solve` **tests**.

Comment (moorepants) [inline sympy/ntheory/continued_fraction.py]: Sorry about all the spam. I didn't look closely at the merged ones.

Thanks for fixing these and leaving any tests for solve in place.

Comment (aktech) [inline sympy/ntheory/continued_fraction.py]: np, there are, a couple of places (merged) `solve` tests, I will take care of those, as well.

File: sympy/ntheory/continued_fraction.py [hunk 2]
```diff
@@ -136,6 +136,7 @@ def continued_fraction_reduce(cf):
     """
     from sympy.core.symbol import Dummy
     from sympy.solvers import solve
```
Comment (gxyd) [inline sympy/ntheory/continued_fraction.py]: Is not this import to be removed ?

Comment (aktech) [inline sympy/ntheory/continued_fraction.py]: Yes, thanks.

-----Original Message-----

**Comment by aktech** [comment]
The `solveset` API may change, so closing this as of now.

**Pull request closed without merging**